template/src/app/signed-in/Settings.tsx

```diff
@@ -60,8 +60,13 @@ function EditProfile(): JSX.Element | null {
 
   async function signOut() {
     setSigningOut(true);
-    await GoogleSignin.signOut();
     await auth().signOut();
+
+    if (Platform.OS === 'web') {
+      await googleWebSignOut();
+    } else {
+      await GoogleSignin.signOut();
+    }
```

Comment:
I think we need to sign out all providers before signing out auth(), otherwise the native SDKs do the auth signout, an auth changed event is fired, a listener is called async, and the user disappears, which causes a main App.tsx re-render which unmounts all these components in a race condition

Reply:
Fixed.